client/src/app/queries/tasks.ts

```diff
@@ -19,56 +19,37 @@ import {
   TaskQueue,
 } from "@app/api/models";
 
-interface FetchTasksFilters {
-  addon?: string;
-  kind?: string;
-}
+export const TaskStates = {
+  Canceled: ["Canceled"],
+  Created: ["Created"],
+  Failed: ["Failed"],
+  Queued: ["Ready", "Postponed", "Pending", "Running"], // "Created", "QuotaBlocked" ??
+  Running: ["Running"],
+  Success: ["Succeeded"],
+};
 
 export const TasksQueryKey = "tasks";
 export const TasksQueueKey = "TasksQueue";
 export const TaskByIDQueryKey = "taskByID";
 export const TaskAttachmentByIDQueryKey = "taskAttachmentByID";
 
-export const useFetchTasks = (
-  filters: FetchTasksFilters = {},
-  refetchDisabled: boolean = false
-) => {
+export const useFetchTasks = (refetchDisabled: boolean = false) => {
   const { isLoading, error, refetch, data } = useQuery({
     queryKey: [TasksQueryKey],
     queryFn: getTasks,
     refetchInterval: !refetchDisabled ? 5000 : false,
     select: (allTasks) => {
-      const uniqSorted = allTasks
-        .filter((task) =>
-          // If there are any tasks with the addon field, we will still need to consider those older
-          // tasks that do not have the kind field. This is because the kind field was added later and is
-          // preferred over the addon field.
-
-          // The task manager will determine and assign the addon field when the addon is specified and addon isnt
-          // which will result in both being set.
-
-          filters?.kind || filters?.addon
-            ? filters.kind === task.kind || filters.addon === task.addon
-            : true
-        )
-        // sort by application.id (ascending) then createTime (newest to oldest)
-        .sort((a, b) =>
-          a.application.id !== b.application.id
-            ? a.application.id - b.application.id
-            : -1 * universalComparator(a.createTime, b.createTime)
-        )
-        // remove old tasks for each application
-        .filter(
-          (task, index, tasks) =>
-            index === 0 ||
-            task.application.id !== tasks[index - 1].application.id
-        );
-
-      return uniqSorted;
+      // sort by createTime (newest to oldest)
+      allTasks.sort(
+        (a, b) => -1 * universalComparator(a.createTime, b.createTime)
+      );
+      return allTasks;
     },
     onError: (err) => console.log(err),
   });
-  const hasActiveTasks = data && data.length > 0;
+
+  const hasActiveTasks =
```

Comment:
For sure to be considered for performance reasons.

Performance wise -- to keep a style of a `DecoratedApplication` around, it may be useful in future that have the state be more selective in updates more like what immerjs and only patch the decorated state model with things that have changed.  :thinking: